File: eMovieFinder/eMovieFinder.Database/Entities/Category.cs
using eMovieFinder.Model.Utilities;
using System.ComponentModel.DataAnnotations;

namespace eMovieFinder.Database.Entities
{
    public class Category : TimeStampObject
    {
        [Key]
        public int Id { get; set; }
        public string CategoryName { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/entities/category.dart
import 'package:json_annotation/json_annotation.dart';

import '../utilities/time_stamp_object.dart';

part 'category.g.dart';

@JsonSerializable()
class Category extends TimeStampObject {
  final int? id;
  final String? categoryName;

  Category({
    super.creationDate,
    super.modifiedDate,
    super.createdById,
    super.modifiedById,
    super.formattedCreationDate,
    super.formattedModifiedDate,
    this.id,
    this.categoryName
  });

  factory Category.fromJson(Map<String, dynamic> json) => _$CategoryFromJson(json);
  Map<String, dynamic> toJson() => _$CategoryToJson(this);
}